src/main/java/lingogo/commons/util/FileUtil.java

```diff
@@ -96,4 +113,78 @@ public static boolean isValidCsvFileName(String fileName) {
+     *
+     * @param filepath A string representing the file path. Cannot be null.
+     * @param expectedHeaders The expected headers in the CSV file.
+     * @throws IOException If an error occurs during the read.
+     * @throws CsvValidationException When the CSV file is not valid.
+     * @throws CsvColumnHeaderException When the headers of the CSV file do not match expectedHeaders.
+     * @throws CsvNumColumnsException When the number of columns in the CSV file does not match the
+     * length of expectedHeaders.
+     */
+    public static List<String[]> importCsvFileContent(String filepath, String[] expectedHeaders)
+            throws IOException, CsvValidationException, CsvColumnHeaderException, CsvNumColumnsException {
+        CSVReader reader = new CSVReaderBuilder(
+                new InputStreamReader(new FileInputStream(filepath), StandardCharsets.UTF_8)).build();
+
+        String[] headers = reader.readNext();
+
+        // Handle empty CSV file
+        if (headers == null) {
+            throw new CsvColumnHeaderException();
+        }
+
+        // Handle UTF-8 byte order mark
+        if (headers.length > 0 && headers[0].startsWith(UTF8_BOM)) {
+            headers[0] = headers[0].substring(1);
+        }
+
+        // Validate headers
+        if (!Arrays.toString(headers).equals(Arrays.toString(expectedHeaders))) {
+            throw new CsvColumnHeaderException();
+        }
+
+        ArrayList<String[]> output = new ArrayList<>();
+        String[] line;
+        int expectedNumberOfColumns = expectedHeaders.length;
+
+        while ((line = reader.readNext()) != null) {
+            if (line.length != expectedNumberOfColumns || Arrays.asList(line).stream().anyMatch(e -> e.isBlank())) {
+                throw new CsvNumColumnsException();
+            }
+
+            output.add(line);
+        }
+
+        return output;
+    }
+
+    /**
+     * Exports the given headers and content into a CSV file at the specified filepath.
+     *
+     * @param filepath The filepath where the CSV file will be created at.
+     * @param headers The headers to be written into the CSV file.
+     * @param content The content to be written into the CSV file. Each element of the List contains a row of data.
+     * @throws IOException When an IO error occurs when writing to the specified file.
+     */
+    public static void exportToCsvFile(String filepath, String[] headers, List<String[]> content)
+            throws IOException {
+        FileOutputStream w = new FileOutputStream(filepath);
+
+        // Add the byte order mark (BOM) for UTF-8 encoding. This is necessary for certain software
+        // (e.g. Excel) to read the CSV file correctly
+        w.write(0xef);
+        w.write(0xbb);
+        w.write(0xbf);
+
+        CSVWriter writer = new CSVWriter(new OutputStreamWriter(w, StandardCharsets.UTF_8));
+
+        writer.writeNext(headers);
+        writer.writeAll(content);
+        writer.close();
```

Comment:
Similar to above comment on the `reader`. In this case, the writer will not be closed if an `IOException` is thrown.